src/utils/bisect.js

```diff
@@ -0,0 +1,97 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+    Copyright (c) 2010/2011 Arnout Kazemier,3rd-Eden
+
+    Permission is hereby granted, free of charge, to any person obtaining a copy
+    of this software and associated documentation files (the "Software"), to deal
+    in the Software without restriction, including without limitation the rights
+    to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+    copies of the Software, and to permit persons to whom the Software is
+    furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included in
+    all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+    AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+    OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+    THE SOFTWARE. 
+*/
+
+// @flow
+
+/**
+ * Calculates the index of the Array where item X should be placed, assuming the Array is sorted.
+ *
+ * @param {Number[] | $ArrayBufferView} array The array containing the items.
+ * @param {Number} x The item that needs to be added to the array.
+ * @param {Number} low Inital Index that is used to start searching, optional.
+ * @param {Number} high The maximum Index that is used to stop searching, optional.
+ * @returns {Number} the index where item X should be placed
+ */
+export function bisectionRight(
+  array: Number[] | $ArrayBufferView,
+  x: Number,
+  low?: Number,
+  high?: Number
+): Number {
```

Comment:
just use `number` instead of `Number` here :) (we don't care for the jsdoc above)